### PR TITLE
Added Google Maps sample code

### PR DESCRIPTION
**Pull request by atsansone**
@domesticmouse : This should cover my Google Maps for Flutter documentation. I am unsure if I have removed all non-necessary files, so I apologize if there are extras included.

### CODE REVIEW

File: google_maps/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
+name: google_maps_in_flutter
+description: A new Flutter project.
+
+# The following line prevents the package from being accidentally published to
```
Comment (domesticmouse) [inline google_maps/pubspec.yaml:4]: Feel free to delete these comments. They add little value in this context.

File: google_maps/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  # The following adds the Cupertino Icons font to your application.
+  # Use with the CupertinoIcons class for iOS style icons.
+  cupertino_icons: ^1.0.2
+  google_maps_flutter: ^2.2.1
+  google_maps_flutter_web: ^0.4.0+3
+  http: ^0.13.5
+  json_annotation: ^4.7.0
+  json_serializable: ^6.5.4
+
+dev_dependencies:
```
Comment (domesticmouse) [inline google_maps/pubspec.yaml:45]: Please add a dependency on `analysis_defaults` as follows:

```yaml
dev_dependencies:
  analysis_defaults:
    path: ../analysis_defaults
```

File: google_maps/analysis_options.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+include: ../../analysis_options.yaml
```
Comment (domesticmouse) [inline google_maps/analysis_options.yaml:1]: Please replace this line with the following:

```yaml
include: package:analysis_defaults/flutter.yaml
```

**Review by domesticmouse — CHANGES_REQUESTED** [review]
Please add this project to the CI pipeline by editing the https://github.com/flutter/samples/blob/main/tool/flutter_ci_script_stable.sh, https://github.com/flutter/samples/blob/main/tool/flutter_ci_script_beta.sh, and https://github.com/flutter/samples/blob/main/tool/flutter_ci_script_master.sh scripts

**Review by atsansone — COMMENTED** [review]
@domesticmouse : Updated!

File: google_maps/pubspec.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+name: google_maps_in_flutter
+description: A new Flutter project.
+publish_to: 'none' 
+version: 1.0.0+1
+environment:
+  sdk: '>=2.18.5 <3.0.0'
```
Comment (domesticmouse) [inline google_maps/pubspec.yaml:6]: 2.19.0 minimum

**Review by domesticmouse — CHANGES_REQUESTED** [review]
Update the minSdk and add this directory to https://github.com/flutter/samples/blob/main/.github/dependabot.yaml to keep deps updated

**Review by domesticmouse — APPROVED** [review]

**Comment by domesticmouse** [comment]
Thanks for doing this @atsansone!

**Pull request merged**